### PR TITLE
Exclude globs

### PR DESCRIPTION
**Pull request by PCSwingle**
Adds -e --exclude option. Merge after #32

### CODE REVIEW

**Review by biobootloader — APPROVED** [review]
@PCSwingle I made some changes, view changes from just my commits [here](https://github.com/biobootloader/mentat/pull/33/files/6a545cdf758c75243208f498e06cae2d7084961f..87690ae74a544c5d456a26329e62b9600897339d).
- made `exclude_paths` optional in run and loop, cleaning up tests.
- changed precedence so the config exclude globs only exclude things added via directories.
- updated readme and configuration.md to clarify

If my changes look good go ahead and merge! 🚀

File: mentat/code_file_manager.py [hunk 1]
```diff
@@ -138,7 +139,11 @@ def _set_file_paths(
             print("Exiting...")
             exit()
 
-        excluded_files = _abs_file_paths_from_list(exclude_paths, check_for_text=False)
+        excluded_files, excluded_files_from_dir = _abs_file_paths_from_list(
+            exclude_paths, check_for_text=False
+        )
+        # excluded paths should have already been expanded
+        assert not excluded_files_from_dir
```
Comment (PCSwingle) [inline mentat/code_file_manager.py:146]: ? Why is this? What if the user tries to --exclude directory?

Comment (biobootloader) [reply mentat/code_file_manager.py:146]: Oh oops. You want to fix?

Comment (PCSwingle) [reply mentat/code_file_manager.py:146]: Yeah i'll get it

**Pull request merged**